### PR TITLE
plan: simplify the selection converting phase.

### PR DESCRIPTION
**Pull request by hanfei1991**
When decide the push down condition in phase of ppd,  then the mained selection must not be pushed down. BuildIndex/TableScan may return a selection plan.

### CODE REVIEW

File: plan/new_physical_plan_builder.go [hunk 1]
```diff
@@ -204,43 +189,6 @@ func (p *baseLogicalPlan) convert2NewPhysicalPlan(prop *requiredProp) (taskProfi
...
-	}
-	task = p.attach2TaskProfile(task)
-	task = prop.enforceProperty(task, p.basePlan.ctx, p.basePlan.allocator)
-	if !prop.isEmpty() {
-		orderedTask, err := p.basePlan.children[0].(LogicalPlan).convert2NewPhysicalPlan(prop)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		orderedTask = p.attach2TaskProfile(orderedTask)
-		// TODO: Here is a trick: selection is the only plan that may not finish the cop task. It is unfair that we compare
```
Comment (winoros) [inline plan/new_physical_plan_builder.go]: This TODO can also be removed?

Comment (hanfei1991) [inline plan/new_physical_plan_builder.go]: Yeah，Selection Plan only contains the condition that can't be pushed

**Comment by winoros** [comment]
LGTM

**Comment by coocood** [comment]
Nice work, much simpler now!
LGTM

**Pull request merged**